src/states/adult.rs

```diff
@@ -76,14 +76,15 @@ pub struct Adult {
 }
 
 impl Adult {
-    pub fn from_joining_peer(
+    pub fn from_peer(
```

Comment:
Minor: maybe just call it `new`? There is no other constructor and the name `from_peer` doesn't provide any additional information.

Reply:
I think it kind of follow the "convention" `into_adult/from_peer`, `into_elder/from_adult`.
But I see your point so will rename from_peer -> new